protocol/rpcconsumer/rpcconsumer.go

```diff
@@ -134,11 +128,21 @@ func (rpcc *RPCConsumer) Start(ctx context.Context, txFactory tx.Factory, client
 				errCh <- err
 				return err
 			}
+
+			// Register For Updates
 			consumerSessionManager := lavasession.NewConsumerSessionManager(rpcEndpoint, optimizer)
 			rpcc.consumerStateTracker.RegisterConsumerSessionManagerForPairingUpdates(ctx, consumerSessionManager)
 
 			finalizationConsensus := &lavaprotocol.FinalizationConsensus{}
 			consumerStateTracker.RegisterFinalizationConsensusForUpdates(ctx, finalizationConsensus)
+
+			err = rpcc.consumerStateTracker.RegisterForSpecUpdates(ctx, chainParser, chainID, rpcEndpoint.ApiInterface)
```

Comment:
we can't do this here, registering for spec updates needs to be before we send the chainParser as an argument to other classes like the optimizer, this will crash on null spec

Reply:
Nice